### PR TITLE
Adding Arrow's Parquet Files to our Tests

### PR DESCRIPTION
**Pull request by pdet**
This PR adds all of arrow's parquet files to our tests for both arrow and the parquet reader, also some small fixes on the arrow integration, mostly related to nested structures.

@hannesmuehleisen I guess the parquet tests mostly concern you, I've created a script that runs parquet files through both our parquet reader and arrow's parquet reader, then I turn arrow's tables into duckdb relations and check if the resulting relation is equal to our parquet scanner.
Then I generate the test/parquet/test_parquet_reader.test, the tests that fail have the mode skip ... mode unskip, so you can have a look later on.
The assumption is that both arrow and our arrow integration are yielding correct results.

### CODE REVIEW

**Comment by hannesmuehleisen** [comment]
Are those files under a compatible license? How many MBs?

**Comment by pdet** [comment]
All parquet files sum up to ~13mb
I got them from here https://github.com/apache/parquet-testing they are Apache-licensed. Since we already had some from this repo, I assume it is compatible?

**Comment by hannesmuehleisen** [comment]
Yes great. Fine by me.

File: test/sql/copy/parquet/test_parquet_remote.test [hunk 1]
```diff
@@ -22,52 +22,52 @@ SELECT * FROM PARQUET_SCAN('https://duckdb.org');
...
-5	Carlos	Burns	cburns4@miitbeian.gov.cn
-6	Kathryn	White	kwhite5@google.com
-7	Samuel	Holmes	sholmes6@foxnews.com
-8	Harry	Howell	hhowell7@eepurl.com
-9	Jose	Foster	jfoster8@yelp.com
-10	Emily	Stewart	estewart9@opensource.org
+#
+## straightforward
+#query IIII
+#SELECT id, first_name, last_name, email FROM PARQUET_SCAN('https://raw.githubusercontent.com/cwida/duckdb/master/data/parquet-testing/userdata1.parquet') LIMIT 10;
```
Comment (Mytherin) [inline test/sql/copy/parquet/test_parquet_remote.test:28]: Is this commented out on purpose?

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! Looks great. Maybe add a LICENSE file next to the files? Otherwise one comment:

File: test/sql/copy/parquet/test_parquet_remote.test (hunk 1 above)
Comment (pdet) [reply test/sql/copy/parquet/test_parquet_remote.test:28]: yeah, the problem is that this is a folder that doesn't exist on the master branch, looking at the link it even uses the old cwida/duckdb I thought this was even moved, no?

Comment (Mytherin) [reply test/sql/copy/parquet/test_parquet_remote.test:28]: That's clearly on purpose to test whether or not the HTTPFS correctly handles HTTP redirections :)

Perhaps (temporarily) link to the older version of those files?

**Comment by pdet** [comment]
I guess not all of our parquet files are from arrow, I can create a new subfolder "arrow" and add a license there.

File: test/sql/copy/parquet/test_parquet_remote.test (hunk 1 above)
Comment (pdet) [reply test/sql/copy/parquet/test_parquet_remote.test:28]: Ha, that makes sense. I overlooked that.

I'll do that then.

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**